fastjsonschema/generator.py

```diff
@@ -164,15 +174,26 @@ def create_variable_keys(self):
         self._variables.add(variable_name)
         self.l('{variable}_keys = set({variable}.keys())')
 
-    def generate_func_code(self, definition):
+    def generate_func_code(self, definition, name):
         """
         Creates base code of validation function and calls helper
         for creating code by definition.
         """
-        with self.l('def func(data):'):
-            self.l('NoneType = type(None)')
+        self._validation_functions_done.add(self._resolver.get_uri())
+        self.l('NoneType = type(None)')
+        self.l('')
+        with self.l('def {}(data):', name):
             self.generate_func_code_block(definition, 'data', 'data')
             self.l('return data')
+        # Generate parts that are referenced and not yet generated
+        while len(self._validation_functions) > 0:
+            uri, name = self._validation_functions.popitem()
```

Comment:
Normal for ... in... loop not works here, because new functions can be found and therefore added in self._validation_functions during it generation, and dictionary cannot changed during iteration.

I add that as comment there. This is rare cases when comment would actually help :)